### PR TITLE
i18n(fr): create `reference/modules/astro-middleware.mdx`

### PR DESCRIPTION
**Pull request by ArmandPhilippot**
#### Description (required)

Moves the Middleware reference from `api-reference.mdx` to a new location `reference/modules/astro-middleware.mdx` (see #9687)

Note: I didn't remove the content in `api-reference.mdx` so at the moment the content is duplicated. I think it's better to do that in a separate PR to avoid conflicts.

#### Related issues & labels (optional)

- Suggested label: i18n

### CODE REVIEW

File: src/content/docs/fr/reference/modules/astro-middleware.mdx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  sequence,
+  createContext,
+  trySerializeLocals,
+  defineMiddleware,
+ } from 'astro:middleware';
+```
+
+### `defineMiddleware()`
+
+You can import and use the utility function `defineMiddleware()` to take advantage of type safety:
```
Comment (thomasbnt) [inline src/content/docs/fr/reference/modules/astro-middleware.mdx:28]:
```suggestion
Vous pouvez importer et utiliser la fonction utilitaire `defineMiddleware()` pour bénéficier tout en sécurité le type :
```

File: src/content/docs/fr/reference/modules/astro-middleware.mdx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+```ts
+// src/middleware.ts
+import { defineMiddleware } from "astro:middleware";
+
+// `context` and `next` are automatically typed
```
Comment (thomasbnt) [inline src/content/docs/fr/reference/modules/astro-middleware.mdx:34]:
```suggestion
// `context` et `next` sont automatiquement saisis
```

**Review by thomasbnt — CHANGES_REQUESTED** [review]
Miss two phrases to translate

File: src/content/docs/fr/reference/modules/astro-middleware.mdx (hunk 1 above)
Comment (ArmandPhilippot) [reply src/content/docs/fr/reference/modules/astro-middleware.mdx:28]: Par contre la fin de la phrase me semble étrange, que penses-tu de :
```suggestion
Vous pouvez importer et utiliser la fonction utilitaire `defineMiddleware()` pour bénéficier de la sûreté du typage :
```

Comment (thomasbnt) [reply src/content/docs/fr/reference/modules/astro-middleware.mdx:28]: > Contributor

Oui c'est mieux!

**Review by thomasbnt — APPROVED** [review]
LGTM! Thanks!

**Review by yanthomasdev — APPROVED** [review]
LGTM!

**Pull request merged**